python_modules/dagster/dagster/_core/definitions/external_asset.py

```diff
@@ -97,6 +97,7 @@ def external_assets_from_specs(specs: Sequence[AssetSpec]) -> List[AssetsDefinit
         )
 
         @multi_asset(
+            name=spec.key.to_python_identifier(),
```

Comment:
this avoids collisions caused when multiple asset specs are provided on the `_external_assets_def` key